### PR TITLE
Update snippet and improve CI

### PR DESCRIPTION
**Pull request by silesky**
- Remove final newline
- Add Linting and CI
- Refactor so the first part of the minified snippet is 'analytics' again 

Will do a minor version bump (5.1.0)

### CODE REVIEW

**Review by oscb — APPROVED** [review]

File: template/snippet.js [hunk 1]
```diff
@@ -116,4 +112,4 @@
   // you'd like to manually name or tag the page, edit or
   // move this call however you'd like.
   '<%= settings.page %>'
-})();
```
Comment (chrisradek) [inline template/snippet.js:119]: What's the reason for this change? It seems to work when I test in the chrome console, just doesn't match the standard IIFE pattern:
https://developer.mozilla.org/en-US/docs/Glossary/IIFE

Comment (silesky) [reply template/snippet.js:119]: Correct -- it works either way. It's part of the default eslint ruleset (more info is here) -- 

https://eslint.org/docs/latest/rules/wrap-iife#:~:text=You%20can%20immediately%20invoke%20function,expression%2C%20rather%20than%20a%20declaration.

Since this is more idiomatic with iifes in 2023, I thought it would make sense to change it.

**Review by chrisradek — APPROVED** [review]
